src/note.cpp

```diff
@@ -8,24 +8,25 @@
 /*************************************************************************/
 
 #include "note.h"
+#include "CommandSet.hpp"
 #include "Descriptor.hpp"
 #include "buffer.h"
 #include "merc.h"
 #include "string_utils.hpp"
-#include "trie.h"
 
 #include <cctype>
 #include <cstdio>
 #include <cstdlib>
 #include <cstring>
 #include <ctime>
+#include <functional>
+#include <memory>
 
 static NOTE_DATA *note_first;
 static NOTE_DATA *note_last;
 
-static void *trie;
-
-typedef void(note_fn_t)(CHAR_DATA *, const char *);
+typedef std::function<void(CHAR_DATA *ch, const char *arg)> note_fn;
+static std::unique_ptr<CommandSet<note_fn>> sub_commands;
```

Comment:
Here why not "just" have an object. It's a global, and yes, construction order is painful, but that's only painful when an object refers to other objects. This does not; so you can just make this `CommandSet<note_fn> sub_commands`

Reply:
Done.